### ISSUE TITLE
Add span-based metric resource

### ISSUE THREAD

**Issue by matjanos** [DataDog/terraform-provider-datadog#1455]
Hi there,

In DataDog UI I am able to create a custom metric generated from Spans (https://docs.datadoghq.com/tracing/generate_metrics/), but I'm not able to do this with Terraform.  

I would love to see this resource in the DataDog provider.

At the moment you are able to create only log-based metric: https://registry.terraform.io/providers/DataDog/datadog/latest/docs/resources/logs_metric

**Comment by tcarrio** [comment]
From what I could gather on this, it appears that there is not an official public API to back this available yet. They're currently using an API nested under a `/ui` path.

Example:

```
POST https://app.datadoghq.com/api/ui/event-platform/spans/custom-metrics

{
  "name": "sample.metric",
  "query": "service:api env:prod",
  "groupBys": [],
  "aggregation": null,
  "enabled": true,
  "_authentication_token": "redacted"
}
```

This route doesn't appear to documented in the Datadog API Docs.

Hopefully this is exposed as a supported public API. Until then I doubt we'll be seeing support in the Terraform provider.

The route does appear to be far simpler than the log-based metrics route (e.g. [Create a log metric](https://docs.datadoghq.com/api/latest/logs-metrics/#create-a-log-based-metric)).

**Comment by kneemaa** [comment]
I am looking for this as well. We want to have some confidence we can recover a defined Span Custom Metric should they be modified or deleted by accident

**Comment by nkzou** [comment]
Closed by #1858 

Thanks for the feature request, a resource has been merged that implements a resource for managing span-based metrics. This will be available for public use on the next release, around the end of the month.
